motor_control/throttle/Button.ino

```diff
@@ -0,0 +1,39 @@
+#include "Button.h"
+
+Button::Button(int ft = 5):Throttle(){
+    isPressed = false;
+    fullThrottle = ft;
+}
+
+int Button::mappingFunction(int f, int s){
+  //linear
+  if(0){
+    return min(s/fullThrottle, 1) * 1023;
+  }
+  //exponential
+  if(1){
+  }
+  //logarithmic
+  if(2){
+  }
```

Comment:
It doesn't look like this function would do anything as is. `if(0)` I imagine would evaluate to `false` and not run, so what is the purpose of this function?

Reply:
I think you are trying to allow for different ramping styles. You can make that an optional parameter that is passed to the ButtonThrottle constructor. Then check `if m_ramp_function==0  { ... }`